napari/layers/base/base.py

```diff
@@ -999,6 +1009,131 @@ def _data_to_world(self) -> Affine:
         """
         return self._transforms[1:3].simplified
 
+    def vector_world_to_data(self, vector) -> tuple:
+        """Convert a vector defining an orientation from world coordinates to data coordinates.
+        For example, this would be used to convert the view ray.
+
+        Parameters
+        ----------
+        vector : tuple, list, 1D array
+            A vector in world coordinates.
+
+        Returns
+        -------
+        tuple
+            Vector in data coordinates.
+        """
+        vector = np.asarray(vector)
+
+        # create transform for view direction (world -> layer data)
+        shear_matrix = expand_upper_triangular(self.shear)
+        rot_matrix = self.rotate
+        scale_matrix = np.diag(self.scale)
+        world_to_layer = np.linalg.inv(
+            shear_matrix @ rot_matrix
+        ).T @ np.linalg.inv(scale_matrix)
+
+        return tuple(vector @ world_to_layer)
+
+    @property
+    def _display_bounding_box(self):
+        """An axis aligned (self._ndisplay, 2) bounding box around the data"""
+        return self._extent_data[:, self._dims_displayed].T
```

Comment:
eek, we have been trying to pull all the `_dims_displayed` off the layer, but i guess this is private and we can deal with it later if we ever actually decide to drop `_dims_displayed`

Reply:
I'm definitely not tied to using `_dims_displayed`. However, to perform selection/picking with nD data, we do need to know which dimensions are displayed. One option could be to have `dims_displayed` on the mouse event. I actually think that is pretty sensible because I don't think it's possible to correctly interpret the mouse event using the mouse event alone without it. What do you think?

Reply:
I have added `dims_displayed` to the mouse event, so our selection code on the layer no long requires `layer._dims_displayed`.